### ISSUE TITLE
Support linting `rq` files

### ISSUE THREAD

**Issue by anderseknert** [open-policy-agent/regal#329]
Regal should be able to lint [rq](https://git.sr.ht/~charles/rq) files "out of the box", meaning that if:

- A user runs `regal lint script.rq`, we should recognize it as Rego with rq additions.
- A user runs `regal lint script` we should recognize it as Rego if the file contents looks like Rego.
- A user runs `regal lint my_dir/` we should be able to recognize rq files [pointed out](https://github.com/StyraInc/regal/issues/328) in config.

I've probably missed some things as I've still more to learn about rq. Note that the scope of this issue is intentionally limited to "support". Once that's in place we can consider rq-_specific_ linter rules.

Related:
- [Support capabilities feature from OPA](https://github.com/StyraInc/regal/issues/212)
- [Allow configuration to decide what files to parse](https://github.com/StyraInc/regal/issues/328)

Tagging @charlesdaniels as a stakeholder :)

**Comment by charlesdaniels** [comment]
Happy to be a resource for this!

I think I have some work to do to expose rq as a Go library. I have created [rq #4](https://todo.sr.ht/~charles/rq/4) to track this work. I have another personal project I also want this for, so this will probably be the next major update to rq.

**Comment by anderseknert** [comment]
Thanks @charlesdaniels! I'm not sure but we probably don't need rq integrated as a library in order to lint its files, we'll just need to be aware of that parsing quirk where the package can be implied. After that it's all AST nodes to Regal.

What would be good though is if the rq project exposed a [capabilities](https://www.openpolicyagent.org/docs/latest/deployments/#capabilities) file where all its built-in functions and their definitions were included. Some linter rules uses the built-in function definitions to determine violations, and these would also be useful if we want rq-specific linter rules further down the line.

**Comment by charlesdaniels** [comment]
Makes sense. Adding a capabilities file is now tracked by [rq #8](https://todo.sr.ht/~charles/rq/8). I do still plan to make rq usable as a library, but good to know it is not a blocker for Regal.

**Comment by anderseknert** [comment]
Found this @charlesdaniels 🙂 I think we can consider this done, even if there may be more to do in the future. Let's add new issues for that if so.
